compiler/rustc_trait_selection/src/traits/query/normalize.rs

```diff
@@ -324,6 +323,70 @@ impl<'cx, 'tcx> FallibleTypeFolder<TyCtxt<'tcx>> for QueryNormalizer<'cx, 'tcx>
                     res
                 }
             }
+
+            ty::Inherent => {
+                // See note in `rustc_trait_selection::traits::project`
```

Comment:
I wonder if we should deduplicate this logic and the one above... This is quite a lot of duplication for what seems to be swapping `tcx.normalize_projection_ty` for `tcx.normalize_inherent_projection_ty`